Mathlib/Combinatorics/Additive/VerySmallDoubling.lean

```diff
@@ -0,0 +1,58 @@
+/-
+Copyright (c) 2024 Yaël Dillies, Patrick Luo. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Yaël Dillies, Patrick Luo
+-/
+import Mathlib.Algebra.Group.Pointwise.Finset.Basic
+import Mathlib.GroupTheory.GroupAction.Defs
+
+/-!
+# Sets with very small doubling
+
+This file characterises sets with no doubling (finsets `A` such that `#(A ^ 2) = #A`) as the sets
+which are either empty or translates of a subgroup.
+
+## TODO
+
+Do we need a version stated using the doubling constant (`Finset.mulConst`)?
```

Comment:
```suggestion
* Do we need a version stated using the doubling constant (`Finset.mulConst`)?
* Add characterisation for sets with doubling < 3/2
```
just so the name makes sense in the meantime

Reply:
Yep, makes sense